ui/app/pages/swaps/swaps-gas-customization-modal/swaps-gas-customization-modal.component.js

```diff
@@ -0,0 +1,232 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import PageContainer from '../../../components/ui/page-container'
+import { Tabs, Tab } from '../../../components/ui/tabs'
+import { calcGasTotal } from '../../send/send.utils'
+import {
+  sumHexWEIsToRenderableFiat,
+} from '../../../helpers/utils/conversions.util'
+import AdvancedGasInputs from '../../../components/app/gas-customization/advanced-gas-inputs'
```

Comment:
We really shouldn't be using sub-components of other components. This is a bit of a red flag. I understand why it's being done here though - this component is between two worlds - the gas customization and swaps.

Maybe this is something we can address at a later date as well.

Ideally what I'd like to see is for us to have a collection of gas-related components  (inputs, gas buttons, the chart, the summary details at the bottom, etc.) that we can mix-and-match as the situation demands, and to have them all be pure-UI with no Redux state, in a directory under `ui/`. With a Storybook file as well. Then we wouldn't have this use of internal components.

Reply:
Created a ticket to capture this https://github.com/MetaMask/metamask-extension/issues/9714